nahusenayElias/Team_4
# DRD-85: post the mautic segment data to JSON:API

## Related ticket: 
[DRD-85](https://edu-team4-react24k.atlassian.net/jira/software/projects/DRD/boards/1?selectedIssue=DRD-85)

## In this PR: 
- made a new content type called Mautic Segment
- enabled automatic fetching of segments from Mautic API to Drupal, meaning that when you create a segment in Mautic, it makes a new Mautic Segment in Drupal automatically. 
- added the Mautic segment checkbox selection to paragraph types. I added it to every paragraph type now. 

## Testing instructions
- checkout branch
- lando db-import DRD-85.gz
- lando drush cim
- lando drush cr
- lando drush cron (this command syncs the segments with Mautic)
- Check from Drupal -> Content that you see the segments in the content list (Content type should be Mautic Segment)
- Try to add a new Blog paragraph (this should be working for other content types using paragraphs too but let's test it with the blog paragraph)
- Check that you see the Mautic Segment checkboxes 
![Screenshot 2024-11-25 at 14-11-11 Create Blog paragraph Drush Site-Install](https://github.com/user-attachments/assets/b65d57ed-66c4-4254-b309-f10f0a9eafa9)

- Create content and choose some segments -> Save

![image](https://github.com/user-attachments/assets/e64ccae8-fce0-416c-9de8-82629acc4aa7)

- Finally, check your JSON:API that you can see the data from the selected segments (you should see segment id and segment name): http://localhost:YOURDRUPALPORT/jsonapi/node/blog_paragraph/NODEID?include=field_paragraph_blog.field_mautic_segments
- Node ID for the jsonapi url can be found in the json api under data when you look at the content you just created in the JSON:API. Data.[0].id in the screenshot below: 
![image](https://github.com/user-attachments/assets/e8d1c8b5-be45-4fad-9a5f-c787c7bba0cd)
- For example, my API endpoint would look like this: http://localhost:63468/jsonapi/node/blog_paragraph/2f85d2e7-27d5-41a9-81f5-20c9e9f64e80?include=field_paragraph_blog.field_mautic_segments
![image](https://github.com/user-attachments/assets/004b652f-986b-4043-9ad5-f0a4329b376b)



[DRD-85]: https://edu-team4-react24k.atlassian.net/browse/DRD-85?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
